src/connection/latestDriverVersionResolver.js

```diff
@@ -9,16 +9,49 @@ function LatestDriverVersionResolver(config, instanceConfig, logger) {
 
 LatestDriverVersionResolver.prototype.getLatestVersion = function (binary) {
   var that = this;
-  return that._getLatestMajorVersion(binary)
+  
+  if (binary.latestVersionUrlRedirect) {
+    return that._getLatestDriverVersion(binary);
+  } else if (binary.latestCompleteVersionFileUrl) {
+    return that._getLatestCompleteVersionFromFile(binary);
+  } else {
+    return that._getLatestMajorVersionFromFile(binary)
     .then(function (result) {
       if (result.latestMajorVersion) {
         binary.latestVersionUrl = binary.latestVersionUrl.replace(that.latestVersionRegexp, result.latestMajorVersion);
       }
       return that._getLatestDriverVersion(binary);
     });
+  }
+};
+
+LatestDriverVersionResolver.prototype._getLatestCompleteVersionFromFile = function (binary) {
+  var that = this;
+
+  that.logger.info('Check for latest complete version of: ' + binary.filename);
+  return q.Promise(function (resolveFn, rejectFn) {
+    if (binary.latestCompleteVersionFileUrl) {
```

Comment:
please move these 'if's to getLatestVersion

Reply:
if (binary.latestCompleteVersionFileUrl)  is in getLatestVersion